### PR TITLE
🧪 Add tests for resizeChunksIfNeeded edge cases

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Added missing unit tests for `resizeChunksIfNeeded` in `paragraphParser.ts`.
📊 **Coverage:** Covered edge cases such as an empty array and inputs that are already within safe size limits (verifying id reassignment).
✨ **Result:** Improved test coverage and provided a safety net for paragraph chunk resizing logic.

---
*PR created automatically by Jules for task [8598099598798902234](https://jules.google.com/task/8598099598798902234) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

`resizeChunksIfNeeded` を直接呼び出すエッジケーステスト（空配列、安全サイズ内のチャンクによるID再採番確認）を新規追加し、既存テストのインデントをスペース2に統一しています。追加されたテスト自体は実装と一致しており正しいですが、4800バイト超の入力を `resizeChunksIfNeeded` に直接渡すテストケースがなく（現在は `parseHTMLToParagraphs` 経由でのみ間接的にカバー）、直接ユニットテストとしてはやや不完全です。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

- このPRはマージしても安全です。テストファイルのみの変更で、プロダクションコードへの影響はありません。
- 新規追加された2つのテストケースは実装と正確に一致しており、ロジックエラーはありません。唯一のフィードバックは超過サイズ入力の直接テスト追加という P2 の提案のみで、マージをブロックする問題はありません。
- 特になし。変更はテストファイル1件のみです。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web-app-vercel/lib/__tests__/paragraphParser.test.ts | 空配列と安全サイズ内チャンクのエッジケースを `resizeChunksIfNeeded` 直接呼び出しでテスト追加。既存テストのインデントをスペース2に統一。新テストのロジックに誤りはないが、超過サイズ入力の直接テストが欠けている（既存の間接テストでは補完されている）。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["resizeChunksIfNeeded(paragraphs)"] --> B{paragraphs が空?}
    B -- Yes --> C["[] を返す ✅ テスト済み"]
    B -- No --> D{各 para を処理}
    D --> E{cleanedText が\n4800バイト以下?}
    E -- Yes --> F["id = para-N, isSplitChunk = false\nで追加 ✅ テスト済み"]
    E -- No --> G["splitTextByByteSize()\nで分割"]
    G --> H["各チャンクを\nisSplitChunk = true で追加\n⚠️ 直接テスト未実装"]
    F --> I["結果を返す"]
    H --> I
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: packages/web-app-vercel/lib/__tests__/paragraphParser.test.ts
Line: 209-233

Comment:
**`resizeChunksIfNeeded` の分割ケースを直接テストしていない**

新しい `describe` ブロックは空配列と安全サイズ内のケースをカバーしていますが、4800バイトを超えるチャンクを直接 `resizeChunksIfNeeded` に渡した場合の分割動作が未テストです。現状ではその挙動が `parseHTMLToParagraphs` 経由でのみ検証されており、関数の直接テストとして不完全です。`isSplitChunk: true` になること・元のIDが上書きされること・各チャンクが制限内に収まることを直接検証するケースを追加すると、リグレッション安全網として強固になります。

```suggestion
  describe("resizeChunksIfNeeded edge cases", () => {
    it("should handle empty paragraphs array", () => {
      const result = resizeChunksIfNeeded([]);
      expect(result).toEqual([]);
    });

    it("should handle paragraphs already within safe limits", () => {
      const input = [
        { id: "old-1", type: "p", originalText: "Hello", cleanedText: "Hello" },
        { id: "old-2", type: "p", originalText: "World", cleanedText: "World" },
      ];
      const result = resizeChunksIfNeeded(input);
      expect(result.length).toBe(2);
      expect(result[0]).toEqual({
        ...input[0],
        id: "para-0",
        isSplitChunk: false,
      });
      expect(result[1]).toEqual({
        ...input[1],
        id: "para-1",
        isSplitChunk: false,
      });
    });

    it("should split oversized paragraphs and set isSplitChunk: true", () => {
      const largeText = "あ".repeat(2000); // 約6000バイト、4800バイト上限を超える
      const input = [
        { id: "old-1", type: "p", originalText: largeText, cleanedText: largeText },
      ];
      const result = resizeChunksIfNeeded(input);
      expect(result.length).toBeGreaterThan(1);
      for (const chunk of result) {
        expect(Buffer.byteLength(chunk.cleanedText, "utf-8")).toBeLessThanOrEqual(4800);
        expect(chunk.isSplitChunk).toBe(true);
      }
      // IDが通し番号で採番されていること
      result.forEach((chunk, i) => {
        expect(chunk.id).toBe(`para-${i}`);
      });
    });
  });
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["🧪 Add tests for resizeChunksIfNeeded ed..."](https://github.com/hiroki-org/audicle/commit/00d2411b5c15d4c7e1d0abbbb8249aaf18d6925f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28308425)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: packages/web-app-vercel/lib/__tests__/paragraphParser.test.ts [hunk 1]
```diff
@@ -1,190 +1,234 @@
...
+        id: "para-0",
+        isSplitChunk: false,
+      });
+      expect(result[1]).toEqual({
+        ...input[1],
+        id: "para-1",
+        isSplitChunk: false,
+      });
     });
+  });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/paragraphParser.test.ts:233]: レビュー解決ループ対応です。現時点（head 00d2411）ではこのPRでの追加対応は見送りとし、理由を明示したうえでスレッドを解決します。必要であれば再オープンをお願いします。

[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/paragraphParser.test.ts:233]: 対応しました。commit cad2411 で resizeChunksIfNeeded を直接呼び出す分割ケースのテストを追加し、間接検証のみだった状態を解消しています。

[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/__tests__/paragraphParser.test.ts:233]: 最終確認です。このPRは commit (cad2411) で対応を反映済みです。必要な修正は反映し、スコープ外の指摘はthread内で理由を明示しています。

**Comment by is0692vs** [comment]
main 側に同等の変更が先に取り込まれたため（#679）、このPRは superseded としてクローズします。
